mithril-aggregator/src/artifact_builder/cardano_immutable_files_full.rs

```diff
@@ -51,25 +50,47 @@ impl CardanoImmutableFilesFullArtifactBuilder {
         }
     }
 
-    async fn create_snapshot_archive(
+    async fn create_immutables_snapshot_archive(
         &self,
-        beacon: &CardanoDbBeacon,
-        snapshot_digest: &str,
+        base_file_name_without_extension: &str,
     ) -> StdResult<FileArchive> {
-        debug!(self.logger, ">> create_snapshot_archive");
+        debug!(self.logger, ">> create_immutables_snapshot_archive");
 
         let snapshotter = self.snapshotter.clone();
-        let snapshot_name = format!(
-            "{}-e{}-i{}.{}",
-            self.cardano_network, *beacon.epoch, beacon.immutable_file_number, snapshot_digest,
+        let snapshot_name = base_file_name_without_extension.to_owned();
+        // spawn a separate thread to prevent blocking
+        let ongoing_snapshot = tokio::task::spawn_blocking(move || -> StdResult<FileArchive> {
+            snapshotter.snapshot_all_completed_immutables(&snapshot_name)
+        })
+        .await??;
+
+        debug!(
+            self.logger,
+            " > Immutables snapshot created: '{ongoing_snapshot:?}'"
         );
+
+        Ok(ongoing_snapshot)
+    }
+
+    async fn create_ancillary_snapshot_archive(
+        &self,
+        immutable_file_number: ImmutableFileNumber,
+        base_file_name_without_extension: &str,
+    ) -> StdResult<FileArchive> {
+        debug!(self.logger, ">> create_ancillary_snapshot_archive");
+
+        let snapshotter = self.snapshotter.clone();
+        let snapshot_name = format!("ancillary-{base_file_name_without_extension}");
```

Comment:
I guess that this format is better as it will be easy to see files related to the same snapshot when listed alphabetically:
```suggestion
        let snapshot_name = format!("{base_file_name_without_extension}-ancillary");
```

Reply:
Done ✅